src/App.tsx

```diff
@@ -8,37 +8,205 @@ import { PostsList } from './components/PostsList';
+import { User } from './types/User';
+import { getUserPosts } from './api/posts';
+import { Post } from './types/Post';
+import { Comment, CommentData } from './types/Comment';
+import { getPostComments, addNewComment, deleteComment } from './api/comments';
+
+const tempUser = {
+  id: 0,
+  name: '',
+  email: '',
+  phone: '',
+};
+
+const tempPost = {
+  id: 0,
+  userId: 0,
+  title: '',
+  body: '',
+};
+
+type SendDataType = {
+  newComment: CommentData,
+  setIsButtonLoading: (IsButtonLoading: boolean) => void,
+  clearText: () => void,
+};
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User>(tempUser);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [isPostLoading, setIsPostLoading] = useState(false);
+  const [isPostLoadingError, setIsPostLoadingError] = useState(false);
+  const [isNoPosts, setIsNoPosts] = useState(false);
+  const [currentPost, setcurrentPost] = useState<Post>(tempPost);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [isCommentLoading, setIsCommentLoading] = useState(false);
+  const [isCommentLoadingError, setIsCommentLoadingError] = useState(false);
+  const [isNoComments, setIsNoComments] = useState(false);
+  const [isNewCommentForm, setIsNewCommentForm] = useState(false);
+
+  const loadUsers = async () => {
+    const loadedUsers = await getAllUsers();
+
+    if (loadedUsers) {
+      setUsers(loadedUsers);
+    }
+  };
+
+  const loadUserPosts = async (userId: number) => {
+    setUserPosts([]);
+
+    try {
+      const loadedUserPosts = await getUserPosts(userId);
+
+      if (loadedUserPosts.length !== 0) {
+        setUserPosts(loadedUserPosts);
+      } else {
+        setIsNoPosts(true);
+      }
+    } catch {
+      setIsPostLoadingError(true);
+      setUserPosts([]);
+    } finally {
+      setIsPostLoading(false);
+    }
+  };
+
+  const loadPostComments = async (postId: number) => {
+    try {
+      const loadedPostComments = await getPostComments(postId);
+
+      if (loadedPostComments.length !== 0) {
+        setPostComments(loadedPostComments);
+      } else {
+        setIsNoComments(true);
+      }
+    } catch {
+      setIsCommentLoadingError(true);
+    } finally {
+      setIsCommentLoading(false);
+    }
+  };
+
+  const addComment = async (sendData: SendDataType) => {
+    const { newComment, setIsButtonLoading, clearText } = sendData;
+
+    try {
+      const addedComment = await addNewComment({
+        postId: currentPost.id,
+        ...newComment,
+      });
+
+      if (addedComment) {
+        setIsButtonLoading(false);
+        clearText();
+        setPostComments((state) => {
+          return [...state, addedComment];
+        });
+      }
+    } catch {
+      setIsButtonLoading(false);
+      setIsCommentLoadingError(true);
+    }
+  };
+
+  const commentDelete = async (commentId: number) => {
```

Comment:
```suggestion
  const deleteComment = async (commentId: number) => {
```
Let's get used to name functions correctly
![image](https://user-images.githubusercontent.com/96409563/230788414-5d10830c-81ca-4c67-8cb8-fcb69c7846b4.png)